### PR TITLE
EAD Session 6 Assignment

### PR DESCRIPTION
**Pull request by meta-manoj-sharma**

### CODE REVIEW

File: EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/model/Element.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * @author Manoj Sharma
+ * @since Sept 4, 2019 
+ * This is Model Class Of Inventory Elements
+ */
+//@XmlRootElement
+public class Element {
+	String name;
+	int quantity;
+	
+	public Element() {
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/model/Element.java]: What is the significance of super here? Remove if not required.

File: EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/rest/API.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	 * This function is used to insert a Element into Records
+	 * @param element ELement Object in JSON Format
+	 * @return String as message
+	 */
+	@POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public String addSingleElement(Element element) {
+		Service.getInstance().addElement(element);
+        String result = element.getName()+" Saved";
+        //return Response.status(201).entity(result).build();
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/rest/API.java]: Remove un-necessary comments.

File: EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/rest/API.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    public String addElements(String element) {
+		Gson gson=new Gson();
+		TypeToken<List<Element>> token = new TypeToken<List<Element>>() {};
+		List<Element> elements = gson.fromJson(element, token.getType());
+		String inserted="";
+		for(Element e: elements){
+			Service.getInstance().addElement(e);
+			inserted += e.getName()+" Saved\n";
+		}
+        //return Response.status(201).entity(inserted).build();
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/rest/API.java]: Remove unnecessary comments.

File: EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/rest/API.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	 * This function is used to delete a specific element
+	 * @param name element name
+	 * @return String as message
+	 */
+	@DELETE
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("{element}")
+	public String deleteElement(@PathParam("element") String name){
+		List<Element> elements = Service.getInstance().getElements();
+		String result="";
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/rest/API.java]: Add space before and after '='.

File: EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/service/Service.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import com.metacube.demo.model.Element;
+/**
+ * This is service class for operations
+ * This is Singleton class to prevent for duplicate objects and Maintain List Of Elemets 
+ * @author Manoj Sharma 
+ * @since Sept 4, 2019
+ *
+ */
+public class Service {
+	private static Service single_instance = null;
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/model/src/main/java/com/metacube/demo/service/Service.java]: Don't use underscore and follow 'camelCasing'.

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+	 * This function is used to delete a specific element
+	 * @param name element name
+	 * @return String as message
+	 */
+	@DELETE
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("{element}")
+	public String deleteElement(@PathParam("element") String name){
+		List<Element> elements = Service.getInstance().getElements();
+		String result="";
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: Add space before and after '=';

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+	 * This function is used to insert a Element into Records
+	 * @param element ELement Object in JSON Format
+	 * @return String as message
+	 */
+	@POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public String addSingleElement(Element element) {
+		Service.getInstance().addElement(element);
+        String result = element.getName()+" Saved";
+        //return Response.status(201).entity(result).build();
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: remove commented code.

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/model/Element.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ * @author Manoj Sharma
+ * @since Sept 4, 2019 
+ * This is Model Class Of Inventory Elements
+ */
+//@XmlRootElement
+public class Element {
+	String name;
+	int quantity;
+	
+	public Element() {
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/src/com/metacube/demo/model/Element.java]: What is the significance of super here? Remove if unnecessary.

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+    public String addElements(String element) {
+		Gson gson=new Gson();
+		TypeToken<List<Element>> token = new TypeToken<List<Element>>() {};
+		List<Element> elements = gson.fromJson(element, token.getType());
+		String inserted="";
+		for(Element e: elements){
+			Service.getInstance().addElement(e);
+			inserted += e.getName()+" Saved\n";
+		}
+        //return Response.status(201).entity(inserted).build();
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: remove commented code.

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+	
+	/**
+	 * this Function is used to delete elements from records, this will remove all records
+	 * @return String as message
+	 */
+	@DELETE
+	@Produces(MediaType.APPLICATION_JSON)
+	public String deleteAllElements(){
+		List<Element> elements = Service.getInstance().getElements();
+		String result="";
```
Comment (jbhavya) [inline EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: Add space before and after '=';

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java (hunk 6 above)
Comment (meta-manoj-sharma) [reply EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: updated

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java (hunk 7 above)
Comment (meta-manoj-sharma) [reply EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: resolved

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/model/Element.java (hunk 8 above)
Comment (meta-manoj-sharma) [reply EAD-Session-6(1)/RestExample/src/com/metacube/demo/model/Element.java]: updated

File: EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java (hunk 9 above)
Comment (meta-manoj-sharma) [reply EAD-Session-6(1)/RestExample/src/com/metacube/demo/rest/API.java]: updated

**Pull request merged**